src/debugger/buildConfig.ts

```diff
@@ -454,23 +449,33 @@ export class TestingDebugConfigurationFactory {
         );
     }
 
+    private buildDescriptionPath(): string {
+        return path.join(this.buildDirectory, this.artifactFolderForTestKind, "description.json");
+    }
+
     private async isUnifiedTestingBinary(): Promise<boolean> {
         // Toolchains that contain https://github.com/swiftlang/swift-package-manager/commit/844bd137070dcd18d0f46dd95885ef7907ea0697
         // no longer produce a .swift-testing binary, instead we want to use `unifiedTestingOutputPath`.
         // In order to determine if we're working with a unified binary we need to check if the .swift-testing
-        // binary _doesn't_ exist, and if it does exist we need to check that it wasn't built by an old toolchain
-        // and is still in the .build directory. We do this by checking its mtime and seeing if it is after
-        // the `buildStartTime`.
+        // binary was produced by the latest build. If it was then we are not using a unified binary.
 
         // TODO: When Swift 6 is released and enough time has passed that we're sure no one is building the .swift-testing
```

Comment:
Maybe add an issue for this

Reply:
yep, I created https://github.com/swiftlang/vscode-swift/issues/994 to follow up on this